layers/core_checks/cc_external_object.cpp

```diff
@@ -556,3 +556,150 @@ bool CoreChecks::PreCallValidateExportMetalObjectsEXT(VkDevice device, VkExportM
+    // 3. Previous 2 combined
+    // Whenever the memory will be used for an image, VK_EXT_external_memory_metal requires that the allocation
+    // is dedicated to that single resource.
+
+    // Case 1 and partially 3
+    // Since we know Metal textures required to be imported as a dedicated allocation, we can do the check directly here instead
+    // of deferring to the binding call.
+    const auto export_memory_info = vku::FindStructInPNextChain<VkExportMemoryAllocateInfo>(allocate_info.pNext);
+    const auto export_memory_info_nv = vku::FindStructInPNextChain<VkExportMemoryAllocateInfoNV>(allocate_info.pNext);
+    if (export_memory_info || export_memory_info_nv) {
+        VkExternalMemoryHandleTypeFlags handles =
+            export_memory_info ? export_memory_info->handleTypes : export_memory_info_nv->handleTypes;
+        if ((handles == VK_EXTERNAL_MEMORY_HANDLE_TYPE_MTLTEXTURE_BIT_EXT) && dedicated_allocation_info == nullptr) {
+            skip |= LogError(" VUID-VkMemoryAllocateInfo-pNext-00639", device, allocate_info_loc.dot(Field::pNext),
+                             "VK_EXTERNAL_MEMORY_HANDLE_TYPE_MTLTEXTURE_BIT_EXT requires textures to be imported as a dedicated"
+                             "allocation.");
+        }
+    }
+
+    // Case 2 and remaining 3
+    const auto import_memory_metal_info = vku::FindStructInPNextChain<VkImportMemoryMetalHandleInfoEXT>(allocate_info.pNext);
+    if (import_memory_metal_info == nullptr) {
+        return skip;
+    }
+
+    // Ensure user is importing correct type before checking any other VUID since we will assume correct type
+    if ((import_memory_metal_info->handleType != VK_EXTERNAL_MEMORY_HANDLE_TYPE_MTLBUFFER_BIT_EXT) &&
+        (import_memory_metal_info->handleType != VK_EXTERNAL_MEMORY_HANDLE_TYPE_MTLTEXTURE_BIT_EXT) &&
+        (import_memory_metal_info->handleType != VK_EXTERNAL_MEMORY_HANDLE_TYPE_MTLHEAP_BIT_EXT)) {
+        skip |= LogError("VUID-VkImportMemoryMetalHandleInfoEXT-handleType-10410", device,
+                         allocate_info_loc.dot(Struct::VkImportMemoryMetalHandleInfoEXT, Field::handleType), "current value is %s",
+                         string_VkExternalMemoryHandleTypeFlagBits(import_memory_metal_info->handleType));
+        return skip;
+    }
+
+    // Only images require to be created as a dedicated allocation. This allows us to check if the format
+    // used by the image allows for importing such images from external memory. We cannot do that with
+    // buffers since we lack the usage, so it's not possible to know if the device allows import
+    // operations on buffers.
+    if (import_memory_metal_info->handleType == VK_EXTERNAL_MEMORY_HANDLE_TYPE_MTLTEXTURE_BIT_EXT) {
+        if (allocate_info.allocationSize != 0) {
+            skip |= LogError("VUID-VkMemoryAllocateInfo-pNext-10397", device,
+                             allocate_info_loc.dot(Field::allocationSize), "VkMemoryAllocateInfo::allocationSize is %" PRId64,
```

Comment:
```suggestion
                             allocate_info_loc.dot(Field::allocationSize), "is %" PRId64,
```

`allocate_info_loc.dot(Field::allocationSize)` will already print the location